src/connection-manager/auto-dial.ts

```diff
@@ -0,0 +1,148 @@
+import { logger } from '@libp2p/logger'
+import type { PeerStore } from '@libp2p/interface-peer-store'
+import type { ConnectionManager } from '@libp2p/interface-connection-manager'
+import { PeerMap } from '@libp2p/peer-collections'
+import PQueue from 'p-queue'
+import { AUTO_DIAL_CONCURRENCY, AUTO_DIAL_PRIORITY, MIN_CONNECTIONS } from './constants.js'
+
+const log = logger('libp2p:connection-manager:auto-dial')
+
+export interface AutoDialInit {
+  /**
+   * The minimum number of connections below which libp2p will start to dial
+   * peers from the peer book. (default: 0)
+   */
+  minConnections?: number
```

Comment:
MinConnections: 0 is the same as saying, I'm allowed to have zero connections, don't autoDial, correct?

Might be useful to call out that setting this to zero disables it

Reply:
Turns out that default was wrong 🤭  - would much rather generate that info but not sure how.

I've updated [the jsdoc for this setting](https://github.com/libp2p/js-libp2p/blob/fix/refactor-connection-manager/src/connection-manager/index.ts#L39-L44) - if it's still not clear could you suggest an edit?

Reply:
https://github.com/libp2p/js-libp2p/blob/4078082dac944c6feb64e9e45e5275d7f0bf6ae4/src/connection-manager/index.ts#L39-L44 looks much better to me, thanks :)